src/client/opamArg.ml

```diff
@@ -922,38 +928,55 @@ end = struct
         "%s was added in version %s of the opam CLI, \
          but version %s has been requested, which is older."
         flag (OpamCLIVersion.to_string valid_since)
-        (OpamCLIVersion.to_string cli)
+        (string_of_sourced_cli cli)
     in
     `Error (false, msg)
 
+  let previsouly_str removal instead =
+    let previous =
+      string_of_cli_option (OpamCLIVersion.previous removal)
+    in
+    match instead with
+    | Some ist ->
+      Printf.sprintf  ". Use %s instead or %s"
+        (OpamConsole.colorise `bold ist) previous
+    | None -> Printf.sprintf ", %s"  previous
+
   let older_flag_error cli removal instead flags =
     let flag = get_long_form flags in
     let msg =
       Printf.sprintf
         "%s was removed in version %s of the opam CLI, \
          but version %s has been requested%s."
         flag (OpamCLIVersion.to_string removal)
-        (OpamCLIVersion.to_string cli)
-        (let previous =
-           string_of_cli_option (OpamCLIVersion.previous removal)
-         in
-         match instead with
-         | Some ist ->
-           Printf.sprintf  ". Use %s instead or %s"
-             (OpamConsole.colorise `bold ist) previous
-         | None -> Printf.sprintf ", %s"  previous)
+        (string_of_sourced_cli cli)
+        (previsouly_str removal instead)
     in
     `Error (false, msg)
 
+  let deprecated_warning cli removal instead flags =
+    let flag = get_long_form flags in
+    OpamConsole.warning
+      "%s is deprecated since version %s of the opam CLI, \
+       but version %s has been requested%s."
+      flag (OpamCLIVersion.to_string removal)
+      (string_of_sourced_cli cli)
+      (previsouly_str removal instead)
```

Comment:
`snd cli` must be `` `Default``, so `fst cli` must be `OpamCLIVersion.current` so there's no need to display it in the message. In fact, the `cli` parameter isn't needed, beyond possibly asserting that ``snd cli = `Default``?

```suggestion
  let deprecated_warning cli removal instead flags =
    let flag = get_long_form flags in
    OpamConsole.warning
      "%s was deprecated in version %s of the opam CLI%s."
      flag (OpamCLIVersion.to_string removal)
      (previously_str removal instead)
```